File: 1642-water-bottles/water-bottles.ts
function numWaterBottles(numBottles: number, numExchange: number): number {
    let drink = 0, empty = 0;
    while(numBottles > 0) {
        drink += numBottles;
        empty += numBottles;
        numBottles = Math.trunc(empty / numExchange)
        empty %= numExchange
    }
    return drink;
};

File: 1642-water-bottles/water-bottles.rs
impl Solution {
    pub fn num_water_bottles(num_bottles: i32, num_exchange: i32) -> i32 {
        let mut total = num_bottles;
        let mut remaining = num_bottles;
        while remaining >= num_exchange {
            let new_bottle = remaining / num_exchange;
            remaining += new_bottle - new_bottle * num_exchange;
            total += new_bottle;
        }
        total
    }
}